### ISSUE TITLE
Parsing error: Unexpected character '#' (ES11)

### ISSUE THREAD

**Issue by JavierRSobrino** [eslint/eslint#13117]
With this source code on VSCode:
```

class P {
	#p = 0;
	ppp() {
	  return this.#p;
	}
}
```

It does detect an error that shouldn't be so in ES11:

`Parsing error: Unexpected character '#' `

My `.eslintrc.json` is:

```
{
    "env": {
        "browser": true,
        "node": true,
        "commonjs": true,
        "es2020": true
        //"es6": true
    },
    "extends": "eslint:recommended",
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "parserOptions": {
        "ecmaVersion": 11
    },
    "rules": {
    }
}
```
I have eslint installed globabally:

```
$ eslint --version
v6.8.0
$ node --version
v13.11.0
$ npm --version
6.14.4

VSCODE Version: 1.43.2
```

**Comment by anikethsaha** [comment]
try using `babel-eslint` parser

**Comment by yeonjuan** [comment]
@JavierRSobrino 
The private fields(`#`) proposal is in stage3, And the ESLint default parser didn't support it now.
So, as @anikethsaha mentioned, you can use other parsers (e.g.[babel-eslint](https://github.com/babel/babel-eslint)) to support it.

**Comment by kaicataldo** [comment]
Closing this issue since the question has been answered.

**Comment by greggman** [comment]
Sorry if this is a noob question. Can someone explain what babel-eslint's job is here? Private fields are shipping in Safari and Chrome so it seems like they are past stage3 in reality if not in documents. Will they be added to eslint directly or will they always require babel-eslint?

**Comment by ljharb** [comment]
@greggman stage 3 is when browsers ship; stage 4 is when it’s in the spec (after it’s been shipped in at least 2 impls). As unfortunate as it is, eslint doesn’t support anything until it’s in a spec document, even if it’s already shipped on the web.
